modules/core/02-client/keeper/keeper.go

```diff
@@ -154,6 +154,43 @@ func (k Keeper) IterateConsensusStates(ctx sdk.Context, cb func(clientID string,
 	}
 }
 
+// IterateMetadata provides an iterator over all stored metadata keys in the client store.
+// For each metadata object, it will perform a callback.
+func (k Keeper) IterateMetadata(ctx sdk.Context, storeprefix []byte, cb func(clientID string, key, value []byte) bool) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := storetypes.KVStorePrefixIterator(store, host.KeyClientStorePrefix)
+
+	defer sdk.LogDeferred(ctx.Logger(), func() error { return iterator.Close() })
+	for ; iterator.Valid(); iterator.Next() {
+		path := string(iterator.Key())
+		if strings.Contains(path, host.KeyClientState) {
+			// skip non client state keys
```

Comment:
This should be "skip client state keys", right? drop the "non"

```suggestion
			// skip client state keys
```